src/mjolnir/pbfgraphparser.cc

```diff
@@ -616,14 +616,18 @@ struct graph_callback : public OSMPBF::Callback {
       } */
       else if (tag.first == "official_name" && !tag.second.empty()) {
         w.set_official_name_index(osmdata_.name_offset_map.index(tag.second));
-
       } else if (tag.first == "max_speed") {
-        try {
-          max_speed = std::stof(tag.second);
-          has_max_speed = true;
-          w.set_tagged_speed(true);
-        } catch (const std::out_of_range& oor) {
-          LOG_INFO("out_of_range thrown for way id: " + std::to_string(osmid));
+        if (tag.second == "unlimited") {
```

Comment:
`max_speed` will not be set now and not marked as a tagged speed; therefore, we will assign a default speed based on the road class.  I don't think this is what you should do.  I think you need to somehow still set the `max_speed`.  Maybe set a new key from lua like `unlimited_speed`, but you will have to search for this key first as the keys may not be in order.  see https://github.com/valhalla/valhalla/blob/master/src/mjolnir/pbfgraphparser.cc#L371-L373

Reply:
What speed would you assign to the edge in that case?

Reply:
whatever it is currently being returned from lua.